### PR TITLE
Admin Prismatic Spray fix + misc projectile fixes

### PR DESCRIPTION
**Pull request by Camryn-Buttes**
[BUG]
## About the PR 
This fixes the admin-exclusive version of the prismatic spray spell to actually be castable.

It also fixes a few runtime errors in projectile code that I encountered while testing.

## Why's this needed? 

Fixes some bugs and reduces the number of runtimes that happen.

### CODE REVIEW

File: code/datums/abilities/wizard.dm [hunk 1]
```diff
@@ -299,7 +299,7 @@
 		object.owner = src
 
 	castcheck()
-		return !istype(src, /datum/targetable/spell/prismatic_spray/admin) && holder.owner.wizard_castcheck(src)
+		return holder.owner.wizard_castcheck(src)
```
Comment (pali6) [inline code/datums/abilities/wizard.dm:302]: Removing this is good but shouldn't you reintroduce this behaviour back in `/datum/targetable/spell/prismatic_spray/admin`'s castcheck? (Basically just overriding it to `return TRUE`?)

Comment (Camryn-Buttes) [reply code/datums/abilities/wizard.dm:302]: I thought about it, but, frankly, it seemed unnecessary in hindsight

Comment (Camryn-Buttes) [reply code/datums/abilities/wizard.dm:302]: [Having it be robeless, that is]

Comment (Camryn-Buttes) [reply code/datums/abilities/wizard.dm:302]: Because we already have a way to bypass that through the arcane_power bioeffect (with variant set to 2)

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**